packages/firestore/test/unit/util/async_queue.test.ts

```diff
@@ -203,6 +203,85 @@ describe('AsyncQueue', () => {
     expect(completedSteps).to.deep.equal([1, 2, 3, 4]);
   });
 
+  it('Retries retryable operations', async () => {
+    const queue = new AsyncQueue();
+    const completedSteps: number[] = [];
+    const doStep = (n: number): void => {
+      completedSteps.push(n);
+    };
+    queue.enqueueRetryable(async () => {
+      doStep(1);
+      if (completedSteps.length === 1) {
+        throw new Error("Let's retry");
```

Comment:
When combing through through test logs, it's useful for simulated errors to clearly label themselves so they can be easily ignored. Consider prefixing this message with "Simulated error" or similar.